### ISSUE TITLE
Strings with a number and whitespaces are considered numeric

### ISSUE THREAD

**Issue by vladshcherbin** [fabiospampinato/is-numeric#1]
Hey, should a string with whitespaces be numeric:

```js
import isNumeric from '@fabiospampinato/is-numeric'
console.log(isNumeric('   3 ')) // -> true
```

I'd expect it to be false.

**Comment by fabiospampinato** [comment]
I'm not 100% sure about this, but I think this function should basically check if you can get a finite number back when doing `Number ( x )`, while also excluding obvious not-numbers, i.e.: 

```javascript 
Number ( [] ) // => 0
```

In this case:

```javascript
Number ( '   3 ' ) // => 3
```

and I'd say `'   3 '` is not an obvious not-number, so I'd argue `isNumeric ( '   3 ' )` should return `true`.

If you can point to a more authoritative `isNumeric` implementation, I might consider aligning to that.

**Comment by vladshcherbin** [comment]
@fabiospampinato yep, you're probably right.

I was fooled by the differences of what is considered `numeric` in different languages. I checked this in php with native `is_numeric` function and the result was `false`. A bit later I came back and checked [this two versions](https://repl.it/repls/WeeInfamousAsianpiedstarling) and remembered how retarded php is 🤦‍♂️

Thanks and sorry for bothering.
